labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -77,7 +90,8 @@ object RecursionPatternMatchingExercise {
    * List(List(1), List('A, 'B, 'C), List('a, 'b)) -> List(List(1, 'A, 'a))
    */
   def zipMultipleWithDifferentSize(in: List[List[_]]): List[List[_]] = {
-    error("fix me")
+    val minSize = in.foldLeft(in(0).size)((x, i) => if (i.size < x) i.size else x)
```

Comment:
Nice. I think it's often clearer when you separate the transformation steps a bit more though, for example `val minSize = in map (_.size) minimum` or similar.